streaming/src/main/scala/org/apache/spark/streaming/scheduler/ReceiverTracker.scala

```diff
@@ -440,7 +430,10 @@ class ReceiverTracker(ssc: StreamingContext, skipReceiverLaunch: Boolean = false
       rcvr
     }
 
-    runDummySparkJob()
+    while ((System.currentTimeMillis() - createTime) < maxRegisteredWaitingTimeMs) {}
```

Comment:
You can't spin on a condition like this; it'll waste CPU in millions of system calls. This also forces a delay of this waiting time, which is not OK.